Python/chapter04/p01_route_between_nodes/miguelHx.py

```diff
@@ -0,0 +1,202 @@
+            node.print_children()
+
+    def reset_visited(self):
+        for node in self.nodes:
+            node.visited = False
+
+
+@dataclass
+class Node:
+    id: int
+    children: 'List[Node]'
+    visited: bool = False
+
+    def add_child(self, *nodes: 'Node'):
+        for node in nodes:
+            self.children.append(node)
+
+    def print_children(self):
+        message = f'Adjacency list for node ({self.id}): '
+        for child in self.children:
+            message += '{}, '.format(child.id)
+        print(message)
+
+    def __str__(self):
+        return f'Node ({self.id}), visited: {self.visited}'
+
+
+def dfs_search(root: Node) -> List[int]:
+    """Simple DFS.
+    takes in a root, returns a list
+    of ids of the sequence of visited
+    nodes.
+
+    Args:
+        root (Node): starting node
+
+    Returns:
+        List[int]: list of node IDs (i.e. [0, 1, 3])
+    """
+    output = []
+    if root is None:
+        return
+    # print(f'Visiting node ({root.id})')
+    root.visited = True
+    # print(root.children)
+    output.append(root.id)
+    for node in root.children:
+        if not node.visited:
+            output.extend(dfs_search(node))
+    return output
+
+def bfs_search(root: Node) -> List[int]:
+    """Simple BFS.
+    takes in a root, returns a list
+    of ids of the sequence of visited
+    nodes.
+
+    Args:
+        root (Node): starting node
+
+    Returns:
+        List[int]: List[int]: list of node IDs (i.e. [0, 1, 4])
+    """
+    visited_nodes: List[Node] = []
+    visited_nodes.append(root)
+    queue: Deque[Node] = deque()
+    root.visited = True
+    queue.append(root)
+    while len(queue) >= 1:
+        node = queue.popleft()
+        node.visited = True
+        # print(f'Visiting node ({node.id})')
+        for n in node.children:
+            if not n.visited:
+                n.visited = True
+                queue.append(n)
+                visited_nodes.append(n)
+    # reset visited state
+    g = Graph(visited_nodes)
+    g.reset_visited()
+    return list(map(lambda n: n.id, visited_nodes))
+
+
+def route_between_nodes(src: Node, dest: Node) -> bool:
+    """This function will return true if a path
+    is found between two nodes, false otherwise.
+    The idea is to perform a breadth first search
+    from src to dest. After obtaining a list of
+    nodes visited, we simply check to see if destination
+    node id is in there.
+
+    Runtime Complexity:
+        O(V + E) where V represents the number of
+        nodes in the graph and E represents the number
+        of edges in this graph.
+    Space Complexity:
+        O(V) where V represents the number of existing nodes
+        in the graph.
+
+    Args:
+        src (Node): from node
+        dest (Node): destination node
+
+    Returns:
+        bool: whether a path between src and dest exists
+    """
+    ids_visited: List[int] = bfs_search(src)
+    return True if dest.id in ids_visited else False
```

Comment:
```suggestion
    return dest.id in ids_visited
```

there's no need for the "ternary"!